crates/polars-core/src/datatypes/static_array.rs

```diff
@@ -19,15 +19,15 @@ pub trait StaticArray: Array {
 }
 
 impl<T: NumericNative> StaticArray for PrimitiveArray<T> {
-    type ValueT<'a> = &'a T;
-    type ValueIterT<'a> = std::slice::Iter<'a, T>;
+    type ValueT<'a> = T;
```

Comment:
It is very inconvenient to have to deal with numeric `&T` where boolean return `bool`. So this makes all copy types return the primitive directly.